discord/state.py

```diff
@@ -281,7 +281,10 @@ def parse_guild_member_remove(self, data):
     def parse_guild_member_update(self, data):
         server = self._get_server(data.get('guild_id'))
         user_id = data['user']['id']
-        member = server.get_member(user_id)
+        try:
+            member = server.get_member(user_id)
+        except AttributeError:
```

Comment:
So looking at this more, the correct thing to check would be to check `if server is None`.  It is impossible for any error to be thrown from the `get_member` method, which means the only other possibility is that the `server` variable was `None`.